lotus/templates/task_instructions.py

```diff
@@ -1,10 +1,102 @@
+from typing import Any
+
 import pandas as pd
 
+from lotus.dtype_extensions import ImageDtype
+
+
+def filter_user_message_formatter(
+    multimodal_data: dict[str, Any] | str,
+    user_instruction: str,
+) -> dict[str, Any]:
+    if isinstance(multimodal_data, str):
+        text = multimodal_data
+        image_inputs: list[dict[str, str]] = []
+    elif isinstance(multimodal_data, dict):
+        image_data: dict[str, str] = multimodal_data.get("image", {})
+        _image_inputs: list[tuple[dict, dict]] = [
+            (
+                {
+                    "type": "text",
+                    "text": f"[{key.capitalize()}]: \n",
+                },
+                {
+                    "type": "image_url",
+                    "image_url": {"url": base64_image},
+                },
+            )
+            for key, base64_image in image_data.items()
+        ]
+        image_inputs = [m for image_input in _image_inputs for m in image_input]
+        text = multimodal_data["text"] or ""
+    else:
+        raise ValueError("multimodal_data must be a dictionary or a string")
+
+    if not image_inputs or len(image_inputs) == 0:
+        return {
+            "role": "user",
+            "content": f"Context:\n{text}\n\nClaim: {user_instruction}",
+        }
+    return {
+        "role": "user",
+        "content": [
+            {
+                "type": "text",
+                "text": f"Claim: {user_instruction}\n\nContext:\n{text}",
+            },
+        ]
+        + image_inputs,
+    }
+
+
+def map_user_message_formatter(
+    multimodal_data: dict[str, Any] | str,
+    user_instruction: str,
+) -> dict[str, Any]:
+    if isinstance(multimodal_data, str):
```

Comment:
There's enough duplicated logic due to the multimodal change at the start of `map_user_message_formatter` and `filter_user_message_formatter` to create a helper function